### PR TITLE
Update Vorpal Name to ASCA(AST-65647)

### PR DESCRIPTION
**Pull request by miryamfoiferCX**
…t practice issues"

By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

> Update Vorpal Name to ASCA.
refactor in user log: Replace "vulnerabilities" by "security best practice violations"

### References

> https://checkmarx.atlassian.net/browse/AST-65647

### Testing

> Describe how this change was tested. Be specific about anything not tested and reasons why. If this solution has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors.
>
> Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

File: src/asca/ascaService.ts [hunk 1]
```diff
@@ -5,16 +5,16 @@ import path from "path";
 import * as os from "os";
 import { error } from "console";
 import { Logs } from "../models/logs";
-import CxVorpal from "@checkmarxdev/ast-cli-javascript-wrapper/dist/main/vorpal/CxVorpal";
+import CxAsca from "@checkmarxdev/ast-cli-javascript-wrapper/dist/main/asca/CxAsca";
```
[the remark this answers is not in the document]
Comment (miryamfoiferCX) [reply src/asca/ascaService.ts:8]: picture
picture2

File: src/asca/ascaService.ts [hunk 2]
```diff
@@ -56,16 +56,17 @@ function ignoreFiles(document: vscode.TextDocument): boolean {
   return document.uri.scheme !== 'file';
 }
 
-export async function clearVorpalProblems() {
+export async function clearAscaProblems() {
   diagnosticCollection.clear();
+  const tempDir = os.tmpdir();
```
[the remark this answers is not in the document]
Comment (miryamfoiferCX) [reply src/asca/ascaService.ts]: sorry, attached by mistake - it was [art of ASCA testing & results in logs...
deleted

File: src/commands/ascaCommand.ts [hunk 3]
```diff
@@ -1,53 +1,54 @@
...
-        .getConfiguration(constants.CheckmarxVorpal)
-        .get(constants.ActivateVorpalAutoScanning) as boolean;
-      if (vorpalActive) {
-        await this.installVorpal();
-        await this.registerVorpalScanOnChangeText();
-        this.logs.info(constants.vorpalStart);
+      const ascaActive = vscode.workspace
+        .getConfiguration(constants.CheckmarxAsca);
+      var ascaActiveAsBool = ascaActive.get(constants.ActivateAscaAutoScanning) as boolean;
+        // .get(constants.ActivateAscaAutoScanning) as boolean;
```
[the remark this answers is not in the document]
Comment (miryamfoiferCX) [reply src/commands/ascaCommand.ts]: refactor. the changes also was part of testing...

File: src/commands/ascaCommand.ts [hunk 4]
```diff
@@ -1,53 +1,54 @@
...
+      if (ascaActiveAsBool.valueOf() === true) {
```
[the remark this answers is not in the document]
Comment (miryamfoiferCX) [reply src/commands/ascaCommand.ts]: as above

**Review by AlvoBen — DISMISSED** [review]

File: src/asca/ascaService.ts [hunk 5]
```diff
@@ -5,16 +5,16 @@ import path from "path";
 import * as os from "os";
 import { error } from "console";
 import { Logs } from "../models/logs";
-import CxVorpal from "@checkmarxdev/ast-cli-javascript-wrapper/dist/main/vorpal/CxVorpal";
+import CxAsca from "@checkmarxdev/ast-cli-javascript-wrapper/dist/main/asca/CxAsca";
 import { constants } from "../utils/common/constants";
 
-const vorpalDir = "CxVorpal";
+const ascaDir = "CxVorpal";
```
[the remark this answers is not in the document]
Comment (miryamfoiferCX) [reply src/asca/ascaService.ts:11]: no
it's the name of the package that we're getting from CLI, and in the cli it's the name of the file that recieving from the engine

File: package.json [hunk 6]
```diff
@@ -882,11 +882,11 @@
         }
       },
       {
-        "title": "Activate Vorpal Real-time Scanning",
-        "id": "vorpal",
+        "title": "Activate ASCA Real-time Scanning",
```
[the remark this answers is not in the document]
Comment (miryamfoiferCX) [reply package.json]: changed

File: .github/workflows/checkmarx-one-scan.yml [hunk 7]
```diff
@@ -9,7 +9,7 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4.1.0
       - name: Checkmarx One CLI Action
-        uses: checkmarx/ast-github-action@6c56658230f79c227a55120e9b24845d574d5225 #2.0.31
+        uses: checkmarx/ast-github-action@9fda5a4a2c297608117a5a56af424502a9192e57 #2.0.34
```
[the remark this answers is not in the document]
Comment (miryamfoiferCX) [reply .github/workflows/checkmarx-one-scan.yml]: done

**Review by OrShamirCM — APPROVED** [review]

**Pull request merged**